pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -124,3 +147,68 @@ func Redis(apimanager *appsv1alpha1.APIManager, client client.Client) (*componen
 	}
 	return component.NewRedis(opts), nil
 }
+
+func (r *RedisReconciler) redisConfigCmMutator(existingObj, desiredObj common.KubernetesObject) (bool, error) {
+	existing, ok := existingObj.(*corev1.ConfigMap)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *v1.ConfigMap", existingObj)
+	}
+	desired, ok := desiredObj.(*corev1.ConfigMap)
+	if !ok {
+		return false, fmt.Errorf("%T is not a *v1.ConfigMap", desiredObj)
+	}
+
+	update := false
+	fieldUpdated := reconcilers.ConfigMapReconcileField(desired, existing, "redis.conf")
+	update = update || fieldUpdated
+
+	return update, nil
+}
+
+func (r *RedisReconciler) isInternalRedis() (bool, error) {
+	if r.apiManager.Spec.ExternalComponents != nil &&
+		r.apiManager.Spec.ExternalComponents.Backend != nil &&
+		r.apiManager.Spec.ExternalComponents.Backend.Redis != nil &&
+		*r.apiManager.Spec.ExternalComponents.Backend.Redis == true {
+		return false, nil
+	}
+	if r.apiManager.Spec.ExternalComponents != nil &&
+		r.apiManager.Spec.ExternalComponents.System != nil &&
+		r.apiManager.Spec.ExternalComponents.System.Redis != nil &&
+		*r.apiManager.Spec.ExternalComponents.System.Redis == true {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (r *RedisReconciler) RollOutRedisPods(ctx context.Context) error {
+
+	podLabelSelector := labels.NewSelector()
+	labelSelector, err := labels.NewRequirement("deployment", selection.In, []string{"backend-redis", "system-redis"})
+	if err != nil {
+		return fmt.Errorf("podlabelSelector failes: %w", err)
+	}
+	podLabelSelector = podLabelSelector.Add(*labelSelector)
+
+	redisPods := &corev1.PodList{}
+	opts := client.ListOptions{
+		Namespace:     r.apiManager.GetNamespace(),
+		LabelSelector: podLabelSelector,
+	}
+	err = r.Client().List(ctx, redisPods, &opts)
+	if err != nil {
+		return fmt.Errorf("failed to list pods: %w", err)
+	}
+
+	if len(redisPods.Items) < 2 {
+		return fmt.Errorf("redis pods not found: %w", err)
+	}
+	for _, pod := range redisPods.Items {
+		err = r.Client().Update(context.Background(), &pod)
```

Comment:
@valerymo I'm trying to understand what are we doing here, it seems to me we are fetching the pod and updating it with the same contents as we just fetched, what am I missing?

Reply:
@MStokluska Thank you for this comment. Yes, `RollOutRedisPods` method is not needed already. Removed.
Done